### ISSUE TITLE
[Failing test][sig-api-machinery] Servers with support for API chunking should return chunks of results for list calls [Conformance]

### ISSUE THREAD

**Issue by aojea** [kubernetes/kubernetes#125700]
@wojtek-t @p0lyn0mial this seems related to this new failures in CI 

https://storage.googleapis.com/k8s-triage/index.html?test=API%20chunking

> Kubernetes e2e suite [It] [sig-api-machinery] Servers with support for API chunking should return chunks of results for list calls [Conformance]

/cc @liggitt

_Originally posted by @aojea in https://github.com/kubernetes/kubernetes/issues/125405#issuecomment-2189217331_

**Comment by wojtek-t** [comment]
Copying my comment from the PR

Regarding the chunking test - my suspicion is that we have more than one chunking test running at a time and we're somehow including items from more than one namespace in the RemainingItems [hence the flake - it depends whether tests are running in parallel or not].
Not yet understand why exactly though - we will debug and fix (though rather tomorrow than today).

**Comment by wojtek-t** [comment]
OK - I know what is happening.

- listItems will return all objects unless IndexFields includes namespace selector
- namespace selector is only an IndexFields for pods
- the actual filtering is happening based on `key`, so we do filter the items, but we don't compute that when computing RemainingItems

The fix is fairly simple to do (and given we're sorting the items anyway, then just iterating over all items and checking whether the key matches seems good enough).

**Comment by wojtek-t** [comment]
@p0lyn0mial - can you please work on the fix?

**Comment by neolit123** [comment]
i already logged https://github.com/kubernetes/kubernetes/issues/125694 earlier today, but it seemed like a flake.
closing it.

**Comment by Vyom-Yadav** [comment]
@aojea @liggitt @wojtek-t  `Kubernetes e2e suite.[It] [sig-api-machinery] Servers with support for API chunking should support continue listing from the last key if the original version has been compacted away, though the list is inconsistent [Slow] [Conformance]` is also failing:

[6/26/2024, 9:28:33 AM ci-kubernetes-kind-conformance-parallel-ipv6](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel-ipv6/1805812353191645184)
[6/26/2024, 7:48:13 AM ci-kubernetes-kind-conformance-parallel](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel/1805787438916833280)
[6/26/2024, 7:28:02 AM ci-kubernetes-kind-conformance-parallel-ipv6](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-kind-conformance-parallel-ipv6/1805782153624031232)
[6/26/2024, 5:49:16 AM periodic-kubernetes-e2e-kind-kms](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-kubernetes-e2e-kind-kms/1805757491636604928)
[6/26/2024, 4:33:41 AM ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-28-on-latest](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-kubeadm-kinder-kubelet-1-28-on-latest/1805738365190934528)

Should I create separate issue for this? Failure reason is different.

---

`Kubernetes e2e suite: [It] [sig-api-machinery] Servers with support for API chunking should return chunks of results for list calls [Conformance]`
```
{ failed [FAILED] Expected
    <int>: 800
to be ==
    <int>: 400
In [It] at: k8s.io/kubernetes/test/e2e/apimachinery/chunking.go:109 @ 06/25/24 18:53:23.85
}
```

---

`Kubernetes e2e suite: [It] [sig-api-machinery] Servers with support for API chunking should support continue listing from the last key if the original version has been compacted away, though the list is inconsistent [Slow] [Conformance]`
```
{ failed [FAILED] Expected
    <int>: 401
to be ==
    <int>: 400
In [It] at: k8s.io/kubernetes/test/e2e/apimachinery/chunking.go:161 @ 06/26/24 00:46:39.581
}
```

---

**Comment by Vyom-Yadav** [comment]
Created https://github.com/kubernetes/kubernetes/issues/125716

**Comment by wojtek-t** [comment]
Fixed by reverting the offending PR: https://github.com/kubernetes/kubernetes/pull/125713
